numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -650,7 +650,7 @@ PyUFunc_IsNaTTypeResolver(PyUFuncObject *ufunc,
 {
     if (!PyTypeNum_ISDATETIME(PyArray_DESCR(operands[0])->type_num)) {
         PyErr_SetString(PyExc_TypeError,
-                "ufunc 'isnat' is only defined for datetime and timedelta.");
+                "ufunc 'isnat' is only defined for 'numpy.datetime64' and 'numpy.timedelta64'.");
```

Comment:
I suggest using `np.` instead of the full `numpy.` prefix for the sake of consistency with the proposed changes in [NEP51](https://numpy.org/neps/nep-0051-scalar-representation.html).

```suggestion
                "ufunc 'isnat' is only defined for np.datetime64 and np.timedelta64.");
```